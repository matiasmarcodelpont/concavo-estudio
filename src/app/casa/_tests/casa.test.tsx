import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import CasaConcavo from '@/app/casa/page'

jest.mock('@/data/data.ts', () => ({
  ambientes: [
    {
      slug: 'cocina',
      name: 'Cocina',
    },
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
    },
  ],
  colaboradores: [
    {
      slug: 'elegance',
      name: 'Elegance',
      website: 'www.pisoselegance.es',
      isMain: true,
      description:
        'Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@pisoselegance.com',
      address: 'Avenida del Diseño 123, Ciudad Elegante, España',
    },
    {
      slug: 'luz-viva',
      name: 'Luz Viva',
      website: 'www.luzviva.com',
      isMain: true,
      description:
        'Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@luzviva.com',
      address: 'Calle de la Luz 456, Ciudad Brillante, México',
    },
    {
      slug: 'muebles-arte',
      name: 'Muebles de Arte',
      website: 'www.mueblesdearte.com',
      isMain: false,
      description: null,
      email: null,
      address: 'Calle del Diseño 789, Ciudad Creativa, Argentina',
    },
    {
      slug: 'cocina-design',
      name: 'Cocina Design',
      website: 'www.cocinadesign.com',
      isMain: false,
      description:
        'Especialistas en diseño y equipamiento de cocinas de alta gama. Ofrecemos soluciones personalizadas para crear la cocina de tus sueños, combinando estilo, funcionalidad y calidad.',
      email: null,
      address: null,
    },
  ],
}))

describe('Casa Cóncavo', () => {
  it('renders Casa Cóncavo unchanged', () => {
    const { container } = render(<CasaConcavo />)
    expect(container).toMatchSnapshot()
  })

  it('renders Casa Cóncavo ambientes', () => {
    render(<CasaConcavo />)

    const ambientesList = screen.getByRole('list', { name: 'Lista de Ambientes' })
    expect(ambientesList).toBeInTheDocument()

    const ambientes = within(ambientesList).getAllByRole('listitem')
    expect(ambientes).toHaveLength(2)
  })

  it('renders Casa Cóncavo main Colaboradores', () => {
    render(<CasaConcavo />)

    const mainColaboradoresList = screen.getByRole('list', { name: 'Lista de los principales Colaboradores' })
    expect(mainColaboradoresList).toBeInTheDocument()

    const mainColaboradores = within(mainColaboradoresList).getAllByRole('listitem')
    expect(mainColaboradores).toHaveLength(2)
  })

  it('renders Casa Cóncavo standard Colaboradores', () => {
    render(<CasaConcavo />)

    const standardColaboradoresList = screen.getByRole('list', { name: 'Lista del resto de los Colaboradores' })
    expect(standardColaboradoresList).toBeInTheDocument()

    const standardColaboradores = within(standardColaboradoresList).getAllByRole('listitem')
    expect(standardColaboradores).toHaveLength(2)
  })

  it('renders Casa Cóncavo main Colaboradores', () => {
    render(<CasaConcavo />)

    const mainColaboradoresList = screen.getByRole('list', { name: 'Lista de los principales Colaboradores' })
    expect(mainColaboradoresList).toBeInTheDocument()

    const mainColaboradores = within(mainColaboradoresList).getAllByRole('listitem')
    expect(mainColaboradores).toHaveLength(2)
  })

  it('renders Casa Cóncavo standard Colaboradores', () => {
    render(<CasaConcavo />)

    const standardColaboradoresList = screen.getByRole('list', { name: 'Lista del resto de los Colaboradores' })
    expect(standardColaboradoresList).toBeInTheDocument()

    const standardColaboradores = within(standardColaboradoresList).getAllByRole('listitem')
    expect(standardColaboradores).toHaveLength(2)
  })
})
