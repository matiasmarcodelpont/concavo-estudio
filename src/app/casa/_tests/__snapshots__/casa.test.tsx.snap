// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Casa Cóncavo renders Casa Cóncavo unchanged 1`] = `
<div>
  <main
    class="text-center"
  >
    <section
      class="mb-12"
    >
      <h1
        class="uppercase mt-12 mb-6 italic text-xl sm:text-2xl md:text-3xl hidden"
      >
        Casa Cóncavo
      </h1>
      <ul
        aria-label="Lista de Ambientes"
        class="space-y-4"
      >
        <li>
          <a
            aria-label="Cocina"
            class="relative w-full h-[61.8vw] max-h-[494px] pl-4 flex justify-center items-center"
            href="/casa/cocina"
          >
            <img
              alt="Cocina"
              class="absolute inset-0 w-full h-full object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=3840&q=75"
              srcset="/_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=640&q=75 640w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=750&q=75 750w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=828&q=75 828w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fambientes%2Fcocina%2F0.jpeg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
            <div
              class="absolute inset-0 bg-black opacity-30 z-20"
            />
            <div
              class="relative max-w-[1600px] flex justify-between grow z-30"
            >
              <p
                class="text-xl sm:text-2xl md:text-3xl text-white"
              >
                Cocina
              </p>
              <svg
                class="lucide lucide-chevron-right size-[30px] sm:size-[40px] md:size-[40px]"
                fill="none"
                height="24"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </div>
          </a>
        </li>
        <li>
          <a
            aria-label="Living/Comedor"
            class="relative w-full h-[61.8vw] max-h-[494px] pl-4 flex justify-center items-center"
            href="/casa/living-comedor"
          >
            <img
              alt="Living/Comedor"
              class="absolute inset-0 w-full h-full object-cover"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=3840&q=75"
              srcset="/_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=640&q=75 640w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=750&q=75 750w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=828&q=75 828w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fambientes%2Fliving-comedor%2F0.jpeg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
            <div
              class="absolute inset-0 bg-black opacity-30 z-20"
            />
            <div
              class="relative max-w-[1600px] flex justify-between grow z-30"
            >
              <p
                class="text-xl sm:text-2xl md:text-3xl text-white"
              >
                Living/Comedor
              </p>
              <svg
                class="lucide lucide-chevron-right size-[30px] sm:size-[40px] md:size-[40px]"
                fill="none"
                height="24"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </div>
          </a>
        </li>
      </ul>
    </section>
    <section
      class="mx-4 sm:mx-6 md:mx-8 mb-12"
    >
      <h1
        class="uppercase mt-12 mb-6 italic text-xl sm:text-2xl md:text-3xl"
      >
        Nuestros Colaboradores
      </h1>
      <ul
        aria-label="Lista de los principales Colaboradores"
        class="flex flex-wrap gap-10 sm:gap-12 md:gap-14 lg:gap-16 mb-10 sm:mb-12 md:mb-14 lg:mb-16"
      >
        <li
          class="list-none"
        >
          <div
            class="flex flex-col items-center w-[200px] sm:w-[300px]"
          >
            <div
              class="w-[200px] sm:w-[300px] h-[100px] m-auto relative mb-3"
            >
              <img
                alt="Elegance"
                class="object-contain grayscale"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fcolaboradores%2Felegance.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fcolaboradores%2Felegance.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Felegance.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <p
              class="text-xs text-darkGray mb-3 text-left"
            >
              Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.
            </p>
          </div>
        </li>
        <li
          class="list-none"
        >
          <div
            class="flex flex-col items-center w-[200px] sm:w-[300px]"
          >
            <div
              class="w-[200px] sm:w-[300px] h-[100px] m-auto relative mb-3"
            >
              <img
                alt="Luz Viva"
                class="object-contain grayscale"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <p
              class="text-xs text-darkGray mb-3 text-left"
            >
              Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.
            </p>
          </div>
        </li>
      </ul>
      <ul
        aria-label="Lista del resto de los Colaboradores"
        class="flex flex-wrap gap-10 sm:gap-12 md:gap-14 lg:gap-16 justify-center"
      >
        <li
          class="list-none"
        >
          <div
            class="w-[200px] sm:w-[300px] h-[100px] m-auto relative mb-3"
          >
            <img
              alt="Muebles de Arte"
              class="object-contain grayscale"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fmuebles-arte.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
        <li
          class="list-none"
        >
          <div
            class="w-[200px] sm:w-[300px] h-[100px] m-auto relative mb-3"
          >
            <img
              alt="Cocina Design"
              class="object-contain grayscale"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
